newIDE/app/src/AssetStore/ExtensionStore/ExtensionsSearchDialog.js

```diff
@@ -101,18 +101,25 @@ export default function ExtensionsSearchDialog({
             isInstalling={isInstalling}
             onInstall={async extensionShortHeader => {
               setIsInstalling(true);
-              onInstallExtension(extensionShortHeader);
-              const wasExtensionInstalled = await installDisplayedExtension(
-                i18n,
-                project,
-                eventsFunctionsExtensionsState,
-                extensionShortHeader
-              );
-              if (wasExtensionInstalled && onExtensionInstalled)
-                onExtensionInstalled(extensionShortHeader);
+              try {
+                onInstallExtension(extensionShortHeader);
+                const wasExtensionInstalled = await installDisplayedExtension(
+                  i18n,
+                  project,
+                  eventsFunctionsExtensionsState,
+                  extensionShortHeader
+                );
+                if (wasExtensionInstalled) {
+                  setExtensionWasInstalled(true);
+                  if (onExtensionInstalled)
+                    onExtensionInstalled(extensionShortHeader);
+                  return true;
+                }
 
-              setExtensionWasInstalled(wasExtensionInstalled);
-              setIsInstalling(false);
+                return false;
+              } finally {
+                setIsInstalling(false);
```

Comment:
this is a bit confusing to me, you're returning a value just before, and have a `finally` too. Does it work?
Should we be clearer and call the function `setIsInstalling` on every branch:
- before returning true
- before returning false
- in a catch for the try

Reply:
The code in `finally` is always called, if there is an error or not. Doing that, I make sure that isInstalling comes back to false, that's to say, buttons are re-enabled.

I prefer this way of doing as we do not duplicate lines unnecessarily and it's quite clear to determine which part of the code is called regardless of what happens in that `try`